packages/grafana-ui/src/types/datasource.ts

```diff
@@ -276,6 +278,12 @@ export abstract class DataSourceApi<
    * Used in explore
    */
   languageProvider?: any;
+
+  /**
+   * Can be optionally implemented to allow datasource to be a source of annotations for dashboard. To be visible
+   * in the annotation editor `annotations` capability also needs to be enabled in plugin.json.
+   */
+  annotationQuery?(options: AnnotationQueryRequest<TQuery>): Promise<AnnotationEvent[]>;
```

Comment:
Added this to the DS API as it was missing. Also helps with search for usages in IDE. The typing of `AnnotationQueryRequest<TQuery>` probably isn't perfect but right now there is lots of `any` in the existing datasources so maybe we can change later when needed, but in general seems sensible default right now.